src/client/src/models/block.ts

```diff
@@ -0,0 +1,107 @@
+import _ from 'lodash';
+
+import { blockShape, origin } from '/constants/tetriminos.ts';
+
+class Block {
```

Comment:
I know we should not use Class.
Just POC, will convert this to function.
It's so clean with class tho